### PR TITLE
Add proxy-spanner

### PR DESCRIPTION
**Pull request by hseeberger**

### CODE REVIEW

File: proxy/jdbc/src/main/resources/application.conf [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+//
+//  database = "cloudstate"
+//  database = ${?POSTGRES_DATABASE}
+//
+//  user = "cloudstate"
+//  user = ${?POSTGRES_USERNAME}
+//
+//  password = "cloudstate"
+//  password = ${?POSTGRES_PASSWORD}
+//}
```
Comment (viktorklang) [inline proxy/jdbc/src/main/resources/application.conf:19]:
```suggestion
```

File: tck/src/it/resources/application.conf [hunk 2]
```diff
@@ -1,40 +1,5 @@
 cloudstate-tck.verify = ["Akka + Node.js", "Akka + Java", "Akka + Scala", "Akka + Go", "Akka + Kotlin"]
 cloudstate-tck.combinations = [{
-  name = "Akka + Node.js"
```
Comment (viktorklang) [inline tck/src/it/resources/application.conf:3]: @hseeberger You don't need to remove the combinations anymore, you just edit the `verify`-list above, and the TCK will only verify the named combinations.

File: proxy/jdbc/src/main/resources/application.conf (hunk 1 above)
Comment (hseeberger) [reply proxy/jdbc/src/main/resources/application.conf:19]: Sure thing ;-)

File: tck/src/it/resources/application.conf (hunk 2 above)
Comment (hseeberger) [reply tck/src/it/resources/application.conf:3]: Thanks!

**Comment by marcoderama** [comment]
For the purposes of the preview, are changes required to the proxy built for the spike?  @hseeberger ?

File: proxy/jdbc/src/main/resources/application.conf [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+      plugin = "akka.persistence.spanner.snapshot"
+    }
+
+    spanner {
+      project=${GCP_PROJECT}
+      project=${SPANNER_INSTANCE}
+      project=${SPANNER_DATABASE}
+    }
+  }
+}
```
Comment (pvlugter) [inline proxy/jdbc/src/main/resources/application.conf:17]: This file looks like it shouldn't be here (spanner config under `proxy/jdbc`).

**Comment by hseeberger** [comment]
@pvlugter thanks for the recent updates! I have rebased against the latest master and think this PR is ready for review/merge.

File: proxy/spanner/src/main/scala/io/cloudstate/proxy/spanner/package.scala [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+ */
+
+package io.cloudstate.proxy
+
+import java.time.{Duration => JDuration}
+import scala.concurrent.duration.{Duration, FiniteDuration}
+
+package object spanner {
+
+  // TODO Remove once on Scala 2.13, because Akka then tranisitively depends on a scala-java8-compat version which has these converters.
```
Comment (pvlugter) [inline proxy/spanner/src/main/scala/io/cloudstate/proxy/spanner/package.scala:24]: On Scala 2.13 now, so can be removed

Comment (hseeberger) [reply proxy/spanner/src/main/scala/io/cloudstate/proxy/spanner/package.scala]: Good catch. Removed.

**Comment by pvlugter** [comment]
Added the repo for this to docker hub, so docker images can be automatically published:

https://hub.docker.com/r/cloudstateio/cloudstate-proxy-spanner

(note: `latest` tag is currently the build from master, not the latest release version)

**Comment by marcoderama** [comment]
Is that something we want to be private?  (I think I've asked this before but can't remember...)

**Comment by pvlugter** [comment]
> Is that something we want to be private? (I think I've asked this before but can't remember...)

This is already in Cloudstate OSS, so not sure why the images would be private.

**Comment by marcoderama** [comment]
Of course it is.  I was confused.  Going away...

**Pull request merged**